### DISCUSSION TITLE
Install help

### DISCUSSION THREAD

**Discussion by jmac0027** [zwave-js/zwave-js-ui#2244] [Q&A]
Newbie Stuck on Install

Going by the install guide, I am trying to figure out what exactly I am supposed to replace in the following line:

      _Using a local folder as persistence
      mkdir store
      docker run --rm -it -p 8091:8091 -p 3000:3000 --device=/dev/serial/by-id/insert_stick_reference_here:/dev/zwave \
      -v $(pwd)/store:/usr/src/app/store zwavejs/zwavejs2mqtt:latest_

The instructions say, "Replace /dev/serial/by-id/insert_stick_reference_here (only the first half of the X:Y mapping!) with the serial device for your Z-Wave stick."

Using lsusb  command my device is shown as follows:
        Bus 001 Device 005: ID 0658:0200 Sigma Designs, Inc. Aeotec Z-Stick Gen5 (ZW090) - UZB

Should the line read:
        docker run --rm -it -p 8091:8091 -p 3000:3000 --device=**/dev/serial/by-id/0658:0200**/dev/zwave \ -v $(pwd)/store:/usr/src/app/store zwavejs/zwavejs2mqtt:latest

or

docker run --rm -it -p 8091:8091 -p 3000:3000 --device=**0658:0200**/zwave \ -v $(pwd)/store:/usr/src/app/store zwavejs/zwavejs2mqtt:latest

or something completely different?

What do I replace and with what?  Thank you for anyone who can help.

My device is a raspberry Pi 4 running Pi OS with Homebridge.  

One additional question, what is the best method to run:  Local folder, service or volume?

**Comment by kpine** [comment]
> Should the line read:
> docker run --rm -it -p 8091:8091 -p 3000:3000 --device=**/dev/serial/by-id/0658:0200**/dev/zwave \ -v $(pwd)/store:/usr/src/app/store zwavejs/zwavejs2mqtt:latest

No, that's not correct. Do a `ls /dev/serial/by-id` from a shell prompt and see what it reports. Replace the example with whatever your system shows (it will probably have the strings 0658 and 0200 in the name). It's literally the full path name of a device file.

> One additional question, what is the best method to run: Local folder, service or volume?

The best method for what? Local folder and volumes would be two ways of persisting the necessary cache files and settings. For that the choice is up to you, there's no best. Local folder is easier to access. Not sure what you mean by "service". Using `docker compose` is the easiest (IMO) way to manage the container.

**Reply by jmac0027** [reply]
Thank you!  Your guidance got me a bit further.  I am getting another error. "docker: Error response from daemon: error gathering device information while adding custom device "usb-0658_0200-if00": no such file or directory."

pi@homebridge:/var/lib/homebridge $ sudo ls /dev/serial/by-id/usb-0658_0200-if00
pi@homebridge:/var/lib/homebridge $ sudo docker run --rm -it -p 8091:8091 -p 3000:3000 --device=usb-0658_0200-if00:/dev/zwave -v $(pwd)/store:/usr/src/app/store zwavejs/zwavejs2mqtt:latest
docker: Error response from daemon: error gathering device information while adding custom device "usb-0658_0200-if00": no such file or directory.
pi@homebridge:/var/lib/homebridge $

**Reply by kpine** [reply]
You need to use the full path, `--device=/dev/serial/by-id/usb-0658_0200-if00:/dev/zwave`.

**Reply by jmac0027** [reply]
I appreciate you sticking with me.  I got it!

**Reply by kpine** [reply]
Not sure why you have a `\` in the command, remove that.

Looking again, it appears you copied and pasted the example command and joined the lines. The `\` has to be at the end of a line, if you're using it.

It also looks like you're running the command from the root (`/`) directory, which means your store will be located at `/store`. You might want to find another location for it.

**Reply by jmac0027** [reply]
You were exactly right.  I edited my previous post once I figured out the syntax error.  I'll take your advice on finding a different directory.  Thanks for helping me get this far!

**Comment by jmac0027** [comment]
I've successfully installed zwavejs2mqtt and added my Gen5+ stick thanks to the help of @kpine.  I then was able to add a GE switch.

My question is whether I also need a MQTT broker?  If so, what is recommended? I've read about Eclipse Mosquitto and Node-Red but I don't know ifI need it.

This setup will be used with homebridge via the homebridge_mqttthing plugin on a PI4.

I wish there was a solid guide to understand how everything works and communicates.  

Cheers.

**Reply by robertsLando** [reply]
> My question is whether I also need a MQTT broker? If so, what is recommended? I've read about Eclipse Mosquitto and Node-Red but I don't know ifI need it.

The answer is it depends, you don't have too if you don't need mqtt features. If you only need hass integration just enable websocket server in settings

**Reply by jmac0027** [reply]
I feel dumb asking this but, what functionality is lost by not using a broker? I am looking to simply turn my zwave devices on and off.  No scenes.  This will be through homebridge and zwavejs2mqtt. Not sure what HASS is.  

Sounds like it might be best to use a broker from the beginning in case additional features are needed later?

**Reply by robertsLando** [reply]
@jmac0027 If you wish to turn on/off devices from control panel UI it's not necessary to setup a broker and enable mqtt gateway functionality on z2m. If you wish to do such things using homebridge or node-red or else you have to enable mqtt on z2m settings and setup a broker in order to be able send/receive values
